loaders/src/glTF/2.0/Extensions/KHR_materials_transmission.ts

```diff
@@ -6,6 +6,258 @@ import { IMaterial } from "../glTFLoaderInterfaces";
+        if (!newValues.length) {
+            return;
+        }
+        if (this.disabled) {
+            return;
+        }
+
+        const newOptions = {
+            ...this._options,
+            ...options
+        };
+
+        const oldOptions = this._options;
+        this._options = newOptions;
+
+
+        // If size changes, recreate everything
+        if (newOptions.renderSize !== oldOptions.renderSize) {
+
+            this._setupRenderTargets();
+        }
+    }
+
+    public getOpaqueTarget(): Nullable<Texture> {
+        return this._opaqueRenderTarget;
+    }
+
+    private shouldRenderAsTransmission(material: Nullable<Material>): boolean {
+        if (!material) {
+            return false;
+        }
+        if (material instanceof PBRMaterial && (material.subSurface.isRefractionEnabled)) {
+            return true;
+        }
+        return false;
+    }
+
+    private _addMesh(mesh: AbstractMesh): void {
+        if (mesh instanceof Mesh) {
+            mesh.onMaterialChangedObservable.add(this.onMeshMaterialChanged.bind(this));
+            if (this.shouldRenderAsTransmission(mesh.material)) {
+                this._transparentMeshesCache.push(mesh);
+            } else {
+                this._opaqueMeshesCache.push(mesh);
+            }
+        }
+        if (this._transparentMeshesCache.length == 0) {
+            this.disabled = true;
+        } else {
+            this.disabled = false;
+        }
+    }
+
+    private _removeMesh(mesh: AbstractMesh): void {
+        if (mesh instanceof Mesh) {
+            mesh.onMaterialChangedObservable.remove(this.onMeshMaterialChanged.bind(this));
+            let idx = this._transparentMeshesCache.indexOf(mesh);
+            if (idx !== -1) {
+                this._transparentMeshesCache.splice(idx, 1);
+            }
+            idx = this._opaqueMeshesCache.indexOf(mesh);
+            if (idx !== -1) {
+                this._opaqueMeshesCache.splice(idx, 1);
+            }
+        }
+        if (this._transparentMeshesCache.length == 0) {
+            this.disabled = true;
+        } else {
+            this.disabled = false;
+        }
+    }
+
+    private _parseScene(): void {
+        this._scene.meshes.forEach(this._addMesh.bind(this));
+        // Listen for when a mesh is added to the scene and add it to our cache lists.
+        this._scene.onNewMeshAddedObservable.add(this._addMesh.bind(this));
+        // Listen for when a mesh is removed from to the scene and remove it from our cache lists.
+        this._scene.onMeshRemovedObservable.add(this._removeMesh.bind(this));
+    }
+
+    // When one of the meshes in the scene has its material changed, make sure that it's in the correct cache list.
+    private onMeshMaterialChanged(mesh: AbstractMesh) {
+        if (mesh instanceof Mesh) {
+            let transparent_idx = this._transparentMeshesCache.indexOf(mesh);
+            let opaque_idx = this._opaqueMeshesCache.indexOf(mesh);
```

Comment:
nit: snake case isn't used anywhere in this class except here. seems a bit odd.
nit: let --> const